### PR TITLE
pc - update error handling for frontend / backend interface

### LINKED ISSUE #12: Improve react useBackend error handling
The PR: https://github.com/brownfield-team/kanban-populator/pull/36/files  has some improvements to the error handling for useBackend.

Please consider making these changes in the dsre codebase also.

### PR DESCRIPTION
**Pull request by pconrad**
In this PR, we make adjustments to the useBackend error handling suggested by Seth and Bryan from their work on the dsre codebase.

This makes the error handling and testing a bit less confusing, and a bit easier to work with.

The user gets toasts with helpful messages, and the number of redundant console.error and console.log messages is reduced.

There is jest test coverage, and full mutation coverage.

Fixes #12

### CODE REVIEW

File: frontend/src/tests/pages/UCSBDates/UCSBDatesIndexPage.test.js [hunk 1]
```diff
@@ -130,9 +130,9 @@ describe("UCSBDatesIndexPage tests", () => {
 
         await waitFor(() => { expect(axiosMock.history.get.length).toBeGreaterThanOrEqual(1); });
 
-        const errorMessage = console.error.mock.calls[0][0];
-        expect(errorMessage).toMatch("Error communicating with backend via GET on /api/ucsbdates/all");
-        restoreConsole();
+        // const errorMessage = console.error.mock.calls[0][0];
+        // expect(errorMessage).toMatch("Error communicating with backend via GET on /api/ucsbdates/all");
+        // restoreConsole();
```
Comment (andrewhlu) [inline frontend/src/tests/pages/UCSBDates/UCSBDatesIndexPage.test.js]: nit: do we still need these comments?

File: frontend/src/tests/pages/Students/StudentsIndexPage.test.js [hunk 2]
```diff
@@ -115,8 +115,8 @@ describe("StudentsIndexPage tests", () => {
 
         await waitFor(() => { expect(axiosMock.history.get.length).toBeGreaterThanOrEqual(1); });
 
-        const errorMessage = console.error.mock.calls[0][0];
-        expect(errorMessage).toMatch("Error communicating with backend via GET on /api/students/all");
+        // const errorMessage = console.error.mock.calls[0][0];
+        // expect(errorMessage).toMatch("Error communicating with backend via GET on /api/students/all");
```
Comment (andrewhlu) [inline frontend/src/tests/pages/Students/StudentsIndexPage.test.js]: nit: do we still need these comments?

File: frontend/src/tests/utils/useBackend.test.js [hunk 3]
```diff
@@ -42,27 +37,25 @@ describe("utils/useBackend tests", () => {
...
             var axiosMock = new AxiosMockAdapter(axios);
 
-            axiosMock.onGet("/api/admin/users").reply(404, {});
-
-            const restoreConsole = mockConsole();
+            axiosMock.onGet("/api/admin/users").reply(404, { message: "Error: Request failed with status code 404" });
 
             const { result, waitFor } = renderHook(() => useBackend(
                 ["/api/admin/users"],
+                // Stryker disable next-line StringLiteral : "GET" is the default, so removing it is equivalent
```
Comment (andrewhlu) [inline frontend/src/tests/utils/useBackend.test.js]: Does this line have any effect in a test file? It seems odd to be disabling lines in a test and not in the actual source code.

**Review by andrewhlu — APPROVED** [review]
Did a quick pass through of the changes, looks good to me! One minor question below.

File: frontend/src/tests/pages/Students/StudentsIndexPage.test.js (hunk 2 above)
Comment (pconrad) [reply frontend/src/tests/pages/Students/StudentsIndexPage.test.js]: Thanks! fixed

File: frontend/src/tests/pages/UCSBDates/UCSBDatesIndexPage.test.js (hunk 1 above)
Comment (pconrad) [reply frontend/src/tests/pages/UCSBDates/UCSBDatesIndexPage.test.js]: removed... thanks!

File: frontend/src/tests/utils/useBackend.test.js (hunk 3 above)
Comment (pconrad) [reply frontend/src/tests/utils/useBackend.test.js]: good point.. I think I was doing a global search for 'GET' and putting it everywhere I found it.  I'll remove it here.

File: frontend/src/tests/pages/Students/StudentsIndexPage.test.js (hunk 2 above)
Comment (pconrad) [reply frontend/src/tests/pages/Students/StudentsIndexPage.test.js]: removed.. thanks!

File: frontend/src/tests/utils/useBackend.test.js (hunk 3 above)
Comment (pconrad) [reply frontend/src/tests/utils/useBackend.test.js]: removed

**Pull request merged**